python/mlcroissant/mlcroissant/_src/operation_graph/operations/download.py

```diff
@@ -128,6 +142,17 @@ class Download(Operation):
     def _download_from_http(self, filepath: epath.Path):
         response = requests.get(self.url, stream=True, timeout=10)
         total = int(response.headers.get("Content-Length", 0))
+
+        hash_type = _get_hash_type(self.node)
+
+        if hash_type == "sha256":
+            hash_eng = hashlib.sha256()
+        elif hash_type == "md5":
+            hash_eng = hashlib.md5()
+        else:
+            pass
+            #TODO: What to do if both hash values are given
```

Comment:
This shouldn't happen ([source](https://github.com/mlcommons/croissant/blob/ec5dc76e4c20d28acf681c926799ee763b855146/python/mlcroissant/mlcroissant/_src/structure_graph/nodes/file_object.py#L39)), but I still added a `ValueError` in the function above.

Reply:
I ran a few tests. And it asserts that "at least one hash" is provided, which needs to be changed to "exactly one hash". 

If 2 hashes are provided, the script continues with whatever the following function returns (due to order of if statement it is "md5" currently.

```
def _get_hash(file_object: FileObject):
    if file_object.md5:
        return hashlib.md5()
    elif file_object.sha256:
        return hashlib.sha256()
    else:
        raise ValueError("md5 and sha256 are not specified.")
```